src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -99,6 +99,33 @@ static json_objectt value_set_dereference_stats_to_json(
   return json_result;
 }
 
+optionalt<exprt> value_set_dereferencet::try_add_offset_to_indices(
+  const exprt &expr,
+  const exprt &offset)
+{
+  if(const auto *index_expr = expr_try_dynamic_cast<index_exprt>(expr))
+  {
+    return index_exprt{index_expr->array(),
+                       plus_exprt{index_expr->index(), offset}};
```

Comment:
Checking: 1) Is the `offset` here in terms of array elements, and not bytes? 2) Is it guaranteed to be of the same type as `index_expr->index().type()` (else do `typecast_exprt::conditional_cast(offset, index_expr->index().type())`)?

Reply:
Elements: yes, since this comes from an expression of the form (ptr + offset), i.e. advancing a number of elements. Clarified by renaming the variable to include the units.

Cast: perhaps not, added conditional cast out of caution.